pkg/cli/validate.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"time"
+
+	"github.com/abcxyz/jvs/client-lib/go/client"
+	"github.com/spf13/cobra"
+)
+
+var (
+	flagToken           string
+	flagAllowBreakglass bool
+)
+
+var validateCmd = &cobra.Command{
+	Use:     "validate",
+	Short:   "To validate the given justification token",
+	Example: `validate --token "example token"`,
+	RunE:    runValidateCmd,
+}
+
+func runValidateCmd(cmd *cobra.Command, args []string) error {
+	ctx := context.Background()
+
+	jvsclient, err := client.NewJVSClient(ctx, &client.JVSConfig{
+		Version:         "1",
+		JWKSEndpoint:    cfg.JWKSEndpoint,
+		CacheTimeout:    5 * time.Minute,
+		AllowBreakglass: flagAllowBreakglass,
+	})
+	if err != nil {
+		return err
+	}
+
+	_, err = jvsclient.ValidateJWT(flagToken)
+	if err != nil {
+		return err
+	}
+
+	_, err = cmd.OutOrStdout().Write([]byte("Token is valid"))
```

Comment:
We should print out the parsed token content instead?

Reply:
+1 - I left a similar comment above. Here's how I'm thinking about possible outputs:

**Completely invalid token**

```text
failed to parse token: not a valid jwt
(exit 65)
```

**Valid token but invalid signature**

```text
audiences     foo.bar, zip.zap
breakglass    false
subject       https://service.corp
# ...
valid         false
(exit 74)
```

**Valid breakglass token**

```text
audiences     foo.bar, zip.zap
breakglass    true
subject       https://service.corp
# ...
valid         true
(exit 0)
```

**Valid token**

```text
audiences     foo.bar, zip.zap
breakglass    false
subject       https://service.corp
# ...
valid         true
(exit 0)
```

Reply:
Any lib recommendation? Things like https://github.com/olekukonko/tablewriter might be overkill.

Reply:
Go has a tabwriter package in stdlib iirc. However, it's not a ton of code to write ourselves:

```go
claimsMap, ok := token.AsMap(ctx)
if !ok {
  # ...
}

longest := 0
claims := make(map[string]string, len(claimsMap))
claimsKeys := make([]string, 0, len(claimsMap))
for k, v := range claimsMap {
  if l := len(k); l > longest {
    longest = l
  }
  claimsKeys = append(claimsKeys, k)
  claims[k] = toString(v)
}
sort.Strings(claimsKeys)

spacing := longest + 5 // arbitrary
for _, k := range claimsKeys {
  fmt.Fprint(w, k)
  for i := 0; i < longest-len(k); i++ {
    fmt.Fprint(w, " ")
  }
  fmt.Fprint(w, claims[k])
  fmt.Fprintln(w)
}
```

I'm sure there's a bounds error or something in there, but that's the general idea.

Reply:
I'm absolute fine to start with something primitive.